src/pages/patientView/sampleManager.tsx

```diff
@@ -18,23 +17,71 @@ export function placeArrow(tooltipEl: any) {
 }
 
 
-// sort samples based event, clinical data and id
+// sort samples based on event, clinical data and id
+// 1. based on sample collection data (timeline event)
+// 2. if all cases have derived normalized case types, put primary first
+// 3. natural sort of sample ids
 export function sortSamples(samples: Array<ClinicalDataBySampleId>,
-                            clinicalDataLegacyCleanAndDerived: { [s:string]:any }) {
-    // sort by
-    // 1. based on sample collection data (timeline event) (unimplemented)
-    // 2. if all cases have derived normalized case types, put primary first (unimplemented)
-    // 3. natural sort of sample ids
-    let sortedSampleIDs: string[] = [];
-    sortedSampleIDs = sortedSampleIDs.concat(samples.map((sample) => sample.id)).sort(naturalSort).reverse();
-
-    return _.sortBy(samples, function(sample) {
-        return sortedSampleIDs.indexOf(sample.id);
+                            clinicalDataLegacyCleanAndDerived: { [s:string]:any },
+                            events?: any) {
+    // natural sort
+    let naturalSortedSampleIDs: string[] = [];
+    naturalSortedSampleIDs = naturalSortedSampleIDs.concat(samples.map((sample) => sample.id)).sort(naturalSort);
```

Comment:
TypeScript gave me some error here, if I want to cast to `string[]`. Not sure why, I'll try another time